Add docs for DNS setup (#1466)
FoundationDB/fdb-kubernetes-operator: docs/manual/customization.md

* We currently only support services with the ClusterIP type. These IPs may not be routable from outside the Kubernetes cluster.
* The Service IP space is often more limited than the pod IP space, which could cause you to run out of service IPs.

## Using DNS

> Using Pod IPs has the limitation that Pods might get a new IP address if they are created and sometimes using service IPs is not the right approach.
> FDB supports to use DNS in the cluster file since 7.1 and the operator can make use of that.
> *Note*: That this requires to build the operator image by yourself until we publish an image that contains the 7.1 library as primary library.
> Building the operator by yourself can be achieved with `docker build --build-arg FDB_VERSION=7.1.25 -t foundationdb/fdb-kubernetes-operator .` as an alternative you can add the following customization to inject the 7.1 library and use it as primary library:

Author: ```suggestion
Using Pod IPs has the limitation that Pods might get a new IP address if they are recreated and sometimes using service IPs is not the right approach.
FDB supports to use DNS in the cluster file since 7.1 and the operator can make use of that.
*Note*: This requires the following customization to inject the 7.1 library and use it as primary library (see code example below). As an alternative you can build the operator image by yourself that contains the 7.1 library as the primary library.
Building the operator by yourself can be achieved with `docker build --build-arg FDB_VERSION=7.1.25 -t foundationdb/fdb-kubernetes-operator .`.
```

```yaml
      initContainers:
         ...
         # Install this library in a special location to force the operator to
         # use it as the primary library.
         - name: foundationdb-kubernetes-init-7-1-primary

>            image: foundationdb/foundationdb-kubernetes-sidecar:7.1.35

Author: ```suggestion
           image: foundationdb/foundationdb-kubernetes-sidecar:7.1.25
```

           args:
             - "--copy-library"
             - "7.1"
             - "--output-dir"
             - "/var/output-files/primary"
             - "--init-mode"
           volumeMounts:
             - name: fdb-binaries
               mountPath: /var/output-files
      containers:
         - name: manager
           imagePullPolicy: IfNotPresent
           env:
             - name: LD_LIBRARY_PATH
               value: /usr/bin/fdb/primary/lib
```

> The important parts here is to add an additional init container with the 7.1 version and copy the library into `.../primary`, this library will be used by the operator as primary library.

Author: ```suggestion
The important part here is to add an additional init container with the 7.1 version and copy the library into `.../primary`, this library will be used by the operator as primary library.
```
